urlc175f05cb8_mchllbrt_PermCode_tgz-pJ8-permlib_examples_UniversalSeparableJ8/src/permlib/examples/PropertyStar.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+
+package permlib.examples;
+
+import permlib.PermUtilities;
+import permlib.Permutation;
+import permlib.classes.PermClassInterface;
+import permlib.classes.SimplePermClass;
+import permlib.property.PermProperty;
+
+/**
+ *
+ * @author Michael Albert
+ */
+public class PropertyStar implements PermProperty {
+    
+    private PermClassInterface c;
+    
+    public PropertyStar(PermClassInterface c) {
+        this.c = c;
+    }
+
+    @Override
+    public boolean isSatisfiedBy(Permutation p) {
+        // For each point in p there is are extensions satisfying c which
+        // "lie in the same" block
+        for(int i = 0; i < p.length(); i++) {
+            if (noExtension(p, i)) return false;
+        }
+        return true;
+    }
+
+    @Override
+    public boolean isSatisfiedBy(int[] values) {
+        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
+    }
+
+    private boolean noExtension(Permutation p, int i) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 44-51, lines 52-59 into separate methods.